hyperspy/misc/array_tools.py

```diff
@@ -123,18 +123,19 @@ def linear_bin(s, scale):
     """
 
     shape2 = s.shape
+    newSpectrum = np.zeros(s.shape)
+    newSpectrum = s[:]
 
-    if len(shape) != len(scale):
+    if len(shape2) != len(scale):
```

Comment:
Line 127: The line `newSpectrum = s[:]` entirely overwrites the variable `newSpectrum`, meaning that the previous line doesn't do anything at all. If you wanted to fill `newSpectrum` with the values in the `s` array you would do:
`newSpectrum[:] = s[:]`, but I'm not sure that's necessary at all.

Shouldn't `newSpectrum = s` be enough?